Wikipedia/Code/WMFColumnarCollectionViewLayout.m

```diff
@@ -145,12 +145,9 @@ - (UICollectionViewLayoutInvalidationContext *)invalidationContextForPreferredLa
 
 - (void)invalidateLayoutWithContext:(WMFCVLInvalidationContext *)context {
     assert([context isKindOfClass:[WMFCVLInvalidationContext class]]);
-    if (context.invalidateEverything) {
+    if (context.invalidateEverything || context.invalidateDataSourceCounts) {
```

Comment:
@coreyfloyd when there is no internet, the layout has fewer sections and items in it. After the copy, I need to remove the extraneous sections and items during updateWithInvalidationContext: since they cause a crash when returned to the collection view.